caikit/core/toolkit/factory.py

```diff
@@ -46,7 +46,7 @@ def name(cls) -> str:
         """
 
     @abc.abstractmethod
-    def __init__(self, config: aconfig.Config):
+    def __init__(self, config: aconfig.Config, instance_name: str):
```

Comment:
To support constructibles that are aware of their own name for ID creation, all constructibles must be able to take an `instance_name` argument. If the derived class doesn't need it, it's free to ignore it.